src/python/pants/backend/rust/util_rules/package.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.backend.rust.target_types import RustPackageTarget
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import GenerateTargetsRequest, GeneratedTargets
+from pants.engine.unions import UnionRule, UnionMembership
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class ResolvedRustPackage:
+    pkg_id: str
+
+    target_dir: str
+
+
+class GenerateTargetsFromRustPackageRequest(GenerateTargetsRequest):
+    generate_from = RustPackageTarget
+
+
+@rule(desc="Generate `rust_crate` targets from `rust_package` target", level=LogLevel.DEBUG)
```

Comment:
These target generation rules are unnecessary in this PR.

In the future, they would be used to generate targets for third-party dependency crates used by a Rust package. `rust_crate` would still be used for first-party Rust crates within a Rust package. (At least if this design remains similar to the Go backend.)